doc/tutorial/search/search_solution5.cpp

```diff
@@ -0,0 +1,93 @@
+#include <seqan3/alignment/configuration/all.hpp>
+#include <seqan3/alignment/pairwise/align_pairwise.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/search/algorithm/search.hpp>
+
+using namespace seqan3;
+
+void run_text_single()
+{
+    dna4_vector text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTAACCCGATGAGCTACCCAGTAGTCGAACTGGGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
+    dna4_vector query{"GCT"_dna4};
+    fm_index index{text};
+
+    using it_t = decltype(std::ranges::begin(text));
+
+    debug_stream << "Searching all best hits allowing for 1 error in a single text\n";
+
+    configuration const search_config = search_cfg::max_error{search_cfg::total{1}} |
+                                        search_cfg::mode{search_cfg::all_best};
+    configuration const align_config = align_cfg::edit |
+                                       align_cfg::aligned_ends{align_cfg::seq1_ends_free} |
+                                       align_cfg::result{align_cfg::with_trace};
+
+    auto results = search(index, query, search_config);
+
+    debug_stream << "There are " << results.size() << " hits.\n";
+    debug_stream << "-----------------\n";
+
+    for (auto && pos : results)
+    {
+        size_t start = pos ? pos - 1 : 0;
+        size_t end   = pos + query.size() + 1;
+        auto text_view = view::subrange<it_t, it_t>{std::ranges::begin(text) + start,
+                                                    std::ranges::begin(text) + end};
+
+        for (auto && alignment : align_pairwise(std::make_pair(text_view, query), align_config))
```

Comment:
use `std::pair{text_view, query}` instead of the make_-functions, they are not needed since C++17 anymore.

Reply:
actually use `std::tie()` here, because you don't want to copy